# Качество подгонки модели
датасет `sleep75`

In [1]:
import numpy as np
import pandas as pd
import statsmodels.formula.api as smf

In [3]:
sleep_df = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/sleep75.csv')
mod1 = smf.ols(formula='sleep~totwrk+age', data=sleep_df).fit()
mod1.summary(slim=True)

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  sleep   R-squared:                       0.109
Model:                            OLS   Adj. R-squared:                  0.106
No. Observations:                 706   F-statistic:                     42.93
Covariance Type:            nonrobust   Prob (F-statistic):           2.56e-18
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept   3469.2006     68.118     50.929      0.000    3335.462    3602.939
totwrk        -0.1490      0.017     -8.912      0.000      -0.182      -0.116
age            2.9239      1.397      2.093      0.037       0.182       5.666
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large,  1e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

$R^2=0.109$ Интерпретация: продолжительность сна `sleep` объясняется моделью (т.е. занятостью и возрастом, `totwrk,age`) на 10.9%

$adj.R^2=0.106$ **неинтерпретируем!**

In [5]:
mod2 = smf.ols(formula='sleep~totwrk+age+male+south+marr', data=sleep_df).fit()
mod2.summary(slim=True)

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  sleep   R-squared:                       0.128
Model:                            OLS   Adj. R-squared:                  0.122
No. Observations:                 706   F-statistic:                     20.51
Covariance Type:            nonrobust   Prob (F-statistic):           3.86e-19
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept   3415.3209     75.944     44.971      0.000    3266.215    3564.427
totwrk        -0.1679      0.018     -9.311      0.000      -0.203      -0.133
age            2.7779      1.387      2.003      0.046       0.055       5.501
male          85.0091     34.878      2.437      0.015      16.531     153.488
south        117.3091     40.580      2.891      0.004      37.637     196.982
marr          36.3932     41.732      0.872      0.383     -45.541     118.327
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.17e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

In [6]:
labour_df = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/Labour.csv')
mod3 = smf.ols(formula='np.log(output)~np.log(capital)+np.log(labour)', data=labour_df).fit()
mod3.summary(slim=True)

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:         np.log(output)   R-squared:                       0.838
Model:                            OLS   Adj. R-squared:                  0.837
No. Observations:                 569   F-statistic:                     1462.
Covariance Type:            nonrobust   Prob (F-statistic):          2.64e-224
===================================================================================
                      coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------
Intercept          -1.7115      0.097    -17.697      0.000      -1.901      -1.522
np.log(capital)     0.2076      0.017     12.077      0.000       0.174       0.241
np.log(labour)      0.7148      0.023     30.890      0.000       0.669       0.760
===================================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

$R^2=0.838$: Выпуск объясняется капиталом и трудом на 83.8%

In [7]:
mod4 = smf.ols(formula='np.log(output)~np.log(capital)+np.log(labour)+np.log(wage)', data=labour_df).fit()
mod4.summary(slim=True)

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:         np.log(output)   R-squared:                       0.888
Model:                            OLS   Adj. R-squared:                  0.888
No. Observations:                 569   F-statistic:                     1499.
Covariance Type:            nonrobust   Prob (F-statistic):          1.65e-268
===================================================================================
                      coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------
Intercept          -5.0073      0.221    -22.649      0.000      -5.442      -4.573
np.log(capital)     0.1493      0.015     10.141      0.000       0.120       0.178
np.log(labour)      0.7204      0.019     37.487      0.000       0.683       0.758
np.log(wage)        0.9214      0.058     16.001      0.000       0.808       1.034
===================================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""